src/plugins/controls/public/control_types/range_slider/range_slider_embeddable.tsx

```diff
@@ -0,0 +1,312 @@
+    this.subscriptions.add(dataFetchPipe.subscribe(this.fetchMinMax));
+
+    // build filters when value change
+    this.subscriptions.add(
+      this.getInput$()
+        .pipe(
+          debounceTime(400),
+          distinctUntilChanged((a, b) => isEqual(a.value, b.value)),
+          skip(1) // skip the first input update because initial filters will be built by initialize.
+        )
+        .subscribe(() => this.fetchMinMax())
+    );
+  };
+
+  private getCurrentDataViewAndField = async (): Promise<{
+    dataView: DataView;
+    field: DataViewField;
+  }> => {
+    const { dataViewId, fieldName } = this.getInput();
+    if (!this.dataView || this.dataView.id !== dataViewId) {
+      this.dataView = await this.dataViewsService.get(dataViewId);
+      if (this.dataView === undefined) {
+        this.onFatalError(
+          new Error(RangeSliderStrings.errors.getDataViewNotFoundError(dataViewId))
+        );
+      }
+      this.updateOutput({ dataViews: [this.dataView] });
+    }
+
+    if (!this.field || this.field.name !== fieldName) {
+      this.field = this.dataView.getFieldByName(fieldName);
+      if (this.field === undefined) {
+        this.onFatalError(new Error(RangeSliderStrings.errors.getDataViewNotFoundError(fieldName)));
+      }
+
+      this.updateComponentState({
+        field: this.field,
+        fieldFormatter: this.field
+          ? this.dataView.getFormatterForField(this.field).getConverterFor('text')
+          : (value: string) => value,
+      });
+    }
+
+    return { dataView: this.dataView, field: this.field! };
+  };
+
+  private updateComponentState(changes: Partial<RangeSliderComponentState>) {
+    this.componentState = {
+      ...this.componentState,
+      ...changes,
+    };
+    this.componentStateSubject$.next(this.componentState);
+  }
+
+  private minMaxAgg = (field?: DataViewField) => {
+    const aggBody: any = {};
+    if (field) {
+      if (field.scripted) {
+        aggBody.script = {
+          source: field.script,
+          lang: field.lang,
+        };
+      } else {
+        aggBody.field = field.name;
+      }
+    }
+
+    return {
+      maxAgg: {
+        max: aggBody,
+      },
+      minAgg: {
+        min: aggBody,
+      },
+    };
+  };
+
+  private fetchMinMax = async () => {
+    const { dataView, field } = await this.getCurrentDataViewAndField();
+    this.updateComponentState({ loading: true });
+    this.updateOutput({ loading: true, dataViews: [dataView] });
+    const embeddableInput = this.getInput();
+    const { ignoreParentSettings, fieldName, query, timeRange } = embeddableInput;
+    let { filters = [] } = embeddableInput;
+
+    if (!field) {
+      this.updateComponentState({ loading: false });
+      this.updateOutput({ loading: false, filters: [] });
+      throw fieldMissingError(fieldName);
+    }
+
+    if (timeRange) {
+      const timeFilter = this.dataService.timefilter.createFilter(dataView, timeRange);
```

Comment:
Here and below, you're batching the timeFilter and the filterpill filters. Those can be **individually** turned on and off in https://github.com/elastic/kibana/pull/128090. In the code currently, turning off filters would also turn off the time range.

I think it should be pretty straightforward to split those out!